clustering/buckets.go

```diff
@@ -0,0 +1,95 @@
+package clustering
+
+import "github.com/paulmach/go.geo"
+
+type buckets struct {
+	southWest *geo.Point
+	threshold float64
+	width     int
+	height    int
+	data      [][]int
+}
+
+func newBuckets(bound *geo.Bound, threshold float64) *buckets {
+	maxsize := 4000.0 * 4000.0
+
+	width := bound.Width() / threshold
```

Comment:
Hopefully the bound and width are being calculated correctly (using the right formula to deal with geometry ).